blueprint/core/blueprint_CanvasView.h

```diff
@@ -59,44 +451,34 @@ namespace blueprint
         {
             View::paint(g);
 
+            jassert(ctx);
+            ctx->init();
+
             if (props.contains("onDraw") && props["onDraw"].isMethod())
             {
-                auto* jsContext = new juce::DynamicObject();
-                juce::Path jsPath;
-
-                jsContext->setProperty("__setFillStyle", juce::var::NativeFunction {
-                    [&g, &jsPath](const juce::var::NativeFunctionArgs& args) -> juce::var {
-                        jassert(args.numArguments == 1);
+                std::vector<juce::var> jsArgs {{ctx.get()}};
+                juce::var::NativeFunctionArgs nfArgs (juce::var(), jsArgs.data(), jsArgs.size());
 
-                        auto hex = args.arguments[0].toString();
-                        g.setColour(juce::Colour::fromString(hex));
-
-                        return juce::var();
-                    }
-                });
-
-                jsContext->setProperty("fillRect", juce::var::NativeFunction {
-                    [&g, &jsPath](const juce::var::NativeFunctionArgs& args) -> juce::var {
-                        jassert(args.numArguments == 4);
-
-                        const int x = args.arguments[0];
-                        const int y = args.arguments[1];
-                        const int width = args.arguments[2];
-                        const int height = args.arguments[3];
-
-                        g.fillRect(x, y, width, height);
+                std::invoke(props["onDraw"].getNativeFunction(), nfArgs);
+            }
+            else
+            {
+                jassertfalse;
```

Comment:
Having TypeScript interfaces for component properties would really help in avoiding these kinds of errors. Make it a strongly types constrains that the <Canvas> component requires an "onDraw" prop.

Reply:
Totally agree. I think this particular case makes more sense as a warning, not an assert. Technically your app can still run and function correctly without this prop, you just won't get your draw callbacks